plugins/vpp/ifplugin/descriptor/span.go

```diff
@@ -0,0 +1,204 @@
+	}
+
+	return adapter.NewSpanDescriptor(typedDescr), ctx
+}
+
+// SetInterfaceIndex should be used to provide interface index immediately after
+// the descriptor registration.
+func (d *SpanDescriptor) SetInterfaceIndex(intfIndex ifaceidx.IfaceMetadataIndex) {
+	d.intfIndex = intfIndex
+}
+
+// Validate checks if required filed are not empty.
+func (d *SpanDescriptor) Validate(key string, value *interfaces.Span) error {
+	if value.InterfaceFrom == "" && value.InterfaceTo == "" {
+		return kvs.NewInvalidValueError(ErrSpanWithoutInterface,
+			"interface_from", "interface_to")
+	}
+	if value.InterfaceFrom == "" {
+		return kvs.NewInvalidValueError(ErrSpanWithoutInterface, "interface_from")
+	}
+	if value.InterfaceTo == "" {
+		return kvs.NewInvalidValueError(ErrSpanWithoutInterface, "interface_to")
+	}
+	return nil
+}
+
+// Create configures SPAN.
+func (d *SpanDescriptor) Create(key string, value *interfaces.Span) (metadata interface{}, err error) {
+	ifaceFrom, found := d.intfIndex.LookupByName(value.InterfaceFrom)
+	if !found {
+		err = errors.Errorf("failed to find InterfaceFrom %s", value.InterfaceFrom)
+		d.log.Error(err)
+		return nil, err
+	}
+
+	ifaceTo, found := d.intfIndex.LookupByName(value.InterfaceTo)
+	if !found {
+		err = errors.Errorf("failed to find InterfaceTo %s", value.InterfaceTo)
+		d.log.Error(err)
+		return nil, err
+	}
+
+	var isL2 uint8
+	if value.IsL2 {
+		isL2 = 1
+	}
+
+	err = d.spanHandler.AddSpan(ifaceFrom.SwIfIndex, ifaceTo.SwIfIndex, uint8(value.State), isL2)
+	if err != nil {
+		err = errors.Errorf("failed to add interface span: %v", err)
+		d.log.Error(err)
+		return nil, err
+	}
+
+	return nil, err
+}
+
+// Delete removes SPAN.
+func (d *SpanDescriptor) Delete(key string, value *interfaces.Span, metadata interface{}) error {
+	var err error
+	ifaceFrom, found := d.intfIndex.LookupByName(value.InterfaceFrom)
+	if !found {
+		err = errors.Errorf("failed to find InterfaceFrom %s", value.InterfaceFrom)
+		d.log.Error(err)
+		return err
+	}
+
+	ifaceTo, found := d.intfIndex.LookupByName(value.InterfaceTo)
+	if !found {
+		err = errors.Errorf("failed to find InterfaceTo %s", value.InterfaceTo)
+		d.log.Error(err)
+		return err
+	}
+
+	var isL2 uint8
+	if value.IsL2 {
+		isL2 = 1
+	}
+
+	err = d.spanHandler.DelSpan(ifaceFrom.SwIfIndex, ifaceTo.SwIfIndex, isL2)
+	if err != nil {
+		err = errors.Errorf("failed to delete interface span: %v", err)
+		d.log.Error(err)
+		return err
+	}
+
+	return err
+}
+
+// Retrieve returns all records from VPP SPAN table.
+func (d *SpanDescriptor) Retrieve(correlate []adapter.SpanKVWithMetadata) (retrieved []adapter.SpanKVWithMetadata, err error) {
+	spans, err := d.spanHandler.DumpSpan()
+	if err != nil {
+		d.log.Error(err)
+		return retrieved, err
+	}
+
+	var nameFrom, nameTo string
+	var exists bool
+	for _, s := range spans {
+		nameFrom, _, exists = d.intfIndex.LookupBySwIfIndex(s.SwIfIndexFrom)
+		if !exists {
+			// should this break or error? dunno
```

Comment:
In the very least it should be logged at least at debug level, because it should not normally happen that something points to index we don't know about.

Reply:
omg, totally forgot about this comment in code :flushed:
yes, sure, I'll add debug logs